client/components/ScenarioEditor/index.jsx

```diff
@@ -35,7 +36,12 @@ class ScenarioEditor extends Component {
         this.setState({ categories });
     }
 
-    async getScenarioData() {
+    onChange(event, { name, value }) {
+        this.props.updateEditorMessage('');
+        this.props.setScenario({ [name]: value });
+    }
+
+    async fetchScenario() {
```

Comment:
For a future iteration it would be nice to extrapolate these scenario data functions out of Scenario and Editor. I think @gnarf might be working on ways to do this with Thunk too.